### ISSUE TITLE
Expose savepoints to API

### ISSUE THREAD

**Issue by kyukhin** [tarantool/tarantool#2655]
Right now in-stmt transactions are not working inside generic transactions.
This is due to the fact that Tarantool doesn't allow nested transaction at all.

The solution is to insert savepoint before execution of each SQL statement inside
of transaction and instead of aborting whole transaction - just rollback to such
savepoint.
This will make SQL transactions fully functional.

To do it - need to have save points API available to SQL frontend.

Blocked by #1803

**Comment by kyukhin** [comment]
Partially done, partially this is duplicate of #1803
